lib/OpenQA/Worker/Common.pm

```diff
@@ -334,11 +334,11 @@ sub setup_websocket {
             else {
                 my $err = $tx->error;
                 $ws = undef;
-                warn "Unable to upgrade connection to WebSocket: " . $err->{code} . ". proxy_wstunnel enabled?" if defined $err;
-                if ($err->{code} eq '404') {
-                    # worker id suddenly not known anymore. Abort. If workerid
-                    # is unset we already detected that in api_call
-                    if ($workerid) {
+                if (defined $err) {
```

Comment:
hm. I am not sure if it's a good approach to not handle the else path. We are in error handling already, the request is no websocket one but the error is not defined. When should that be allowed?